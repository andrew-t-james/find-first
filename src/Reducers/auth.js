export const authReducer = (state = {}, action) => {
  switch (action.type) {
    case 'GOOGLE_LOGIN':
      return {
        ...state,
        id: action.user.uid,
        image: action.user.photoURL,
        name: action.user.displayName
      };
    case 'GITHUB_LOGIN':
      return {
        ...state,
        id: action.user.uid,
        image: action.user.photoURL,
        name: action.user.displayName
      };
    case 'TWITTER_LOGIN':
      return {
        ...state,
        id: action.user.uid,
        image: action.user.photoURL,
        name: action.user.displayName
      };
    case 'FACEBOOK_LOGIN':
      return {
        ...state,
        id: action.user.uid,
        image: action.user.photoURL,
        name: action.user.displayName
      };
    case 'GOOGLE_LOGOUT':
      return {};
    default:
      return state;
  }
};