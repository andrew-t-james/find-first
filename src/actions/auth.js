export const googleSignInAction = user => ({
  type: 'GOOGLE_LOGIN',
  id: user.uid,
  name: user.displayName,
  image: user.photoURL
});

export const googleSignOutAction = () => ({
  type: 'GOOGLE_LOGOUT'
});