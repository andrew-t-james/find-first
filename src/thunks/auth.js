export const googleLogin = (authProvider, loginAction) => {
  return async dispatch => {
    const newUser = await authProvider();
    dispatch(loginAction(newUser.user));
  };
};

export const googleLogout = (authLogout, logoutAction) => {
  return async dispatch => {
    await authLogout();
    dispatch(logoutAction());
  };
};