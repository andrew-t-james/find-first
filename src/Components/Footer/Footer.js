import React from 'react';
import { NavLink } from 'react-router-dom';
import { googleLogout } from '../../thunks/auth';
import { logout } from '../../firebase/firebase';
import { googleSignOutAction } from '../../Actions/auth';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const Footer = ({ googleLogout }) => {
  return (
    <footer className="footer">
      <ul className="footer__nav">
        <NavLink className="footer__nav--item" exact to="/dashboard">
          <i className="far fa-clone"></i>
          Dashboard
        </NavLink>
        <NavLink className="footer__nav--item" exact to="/saved-jobs">
          <i className="far fa-user-circle"></i>
          My Jobs
        </NavLink>
        <NavLink className="footer__nav--item" exact to="/sign-in">
          <i className="fas fa-sign-in-alt"></i>
          Sign In
        </NavLink>
        <button
          onClick={() => googleLogout(logout, googleSignOutAction)}
          className="footer__nav--item sign-out"
        >
          <i className="far fa-times-circle"></i>
          Sign Out
        </button>
      </ul>
    </footer>
  );
};


export const mapDispatchToProps = dispatch => ({
  googleLogout: (authLogout, logOutAction) => dispatch(googleLogout(authLogout, logOutAction))
});

export default connect(null, mapDispatchToProps)(Footer);

Footer.propTypes = {
  googleLogout: PropTypes.func
};