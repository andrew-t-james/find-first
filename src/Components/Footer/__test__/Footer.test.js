import React from 'react';
import { shallow } from 'enzyme';
import { Footer, mapDispatchToProps } from '../Footer';
import { googleSignOutAction } from '../../../Actions/auth';

jest.mock('../../../firebase/firebase.js');

describe('<Footer />', () => {
  let wrapper;
  const mockGoogleSignOut = jest.fn();
  beforeEach(() => wrapper = shallow(<Footer googleLogout={mockGoogleSignOut}/>));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call handleGoogleSignOut onClick', () => {
    wrapper.find('.sign-out').simulate('click');

    expect(mockGoogleSignOut).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    test('should call dispatch googleLogout is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = googleSignOutAction();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.googleLogout();

      expect(mockDispatch).toHaveBeenCalled();
    });
  });

});
