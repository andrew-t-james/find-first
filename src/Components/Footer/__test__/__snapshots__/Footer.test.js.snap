// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<footer
  className="footer"
>
  <ul
    className="footer__nav"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="footer__nav--item"
      exact={true}
      to="/dashboard"
    >
      <i
        className="far fa-clone"
      />
      Dashboard
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="footer__nav--item"
      exact={true}
      to="/sign-up"
    >
      <i
        className="far fa-user-circle"
      />
      My Jobs
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="footer__nav--item"
      exact={true}
      to="/sign-in"
    >
      <i
        className="fas fa-sign-in-alt"
      />
      Sign In
    </NavLink>
    <button
      className="footer__nav--item sign-out"
      exact={true}
      onClick={[Function]}
      to="/"
    >
      <i
        className="far fa-times-circle"
      />
      Sign Out
    </button>
  </ul>
</footer>
`;
